### PR TITLE
DataLinks: Fixes datalinks with onClick and variables in url not being interpolated 

### LINKED ISSUE #86242: grafana-data package: Links with onClick method and scopedVars have incorrect href
### What happened?

When using the overrideLinks method with a url containing a data link and an onClick handler, the data link variables don't get replaced in the url. The code block below results in the following screenshot, where I have used Ctrl+Click to open the link in a new tab.

```typescript
.overrideLinks([
  {
    title: 'Go to trace',
    url: BuildUrlParams('explore', '', [
      {
        key: 'panes', value: JSON.stringify({
          'A': {
            datasource: TEMPO_REF,
            queries: [{
              datasource: TEMPO_REF,
              queryType: 'traceql',
              query: '${__value.text}',
            }],
          },
        })
      },
      { key: 'schemaVersion', value: '1' },
    ]),
    onClick: (event: DataLinkClickEvent) => {
      // some onClick handler
    }
  },
])
``` 

image

### What did you expect to happen?

The value of the data link variable should have been interpolated correctly

### Did this work before?

No

### How do we reproduce it?

1. Create a link with both an onClick handler and a url with a data link variable
2. Use Ctrl+Click to open the link in a new tab
3. The data link variable will be displayed without interpolation

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

Grafana: 10.4.1
OS: macOS
Browser: Chrome

### Grafana platform?

Docker

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by gng0**
Added the replaceVariables method to be used on the link url when an onClick handler is provided.
Fixed a unit test and added another test for this fix. More details in the issue

Fixes https://github.com/grafana/grafana/issues/86242

### CODE REVIEW

**Review by torkelo — APPROVED** [review]

**Comment by torkelo** [comment]
Thanks for contributing and fixing this!

**Review by axelavargas — COMMENTED** [review]
@gng0 thank you for contributing to Grafana 🙌🏾 . I see some failures in our CI, can you update your branch with the latest main? 

🙏🏾

**Comment by gng0** [comment]
Done! @axelavargas

**Comment by gng0** [comment]
Hi @torkelo, just wanted to follow up on the continuous-integration/drone/pr build which is pending. The details links me to a page mentioning - "Build is blocked, please, contact repo admin in order to proceed"

File: packages/grafana-data/src/field/fieldOverrides.ts [hunk 1]
```diff
@@ -470,7 +470,7 @@ export const getLinksSupplier =
 
       if (link.onClick) {
         linkModel = {
-          href: link.url,
+          href: replaceVariables(link.url, dataLinkScopedVars, VariableFormatID.UriEncode),
```
Comment (axelavargas) [inline packages/grafana-data/src/field/fieldOverrides.ts:473]: Hey @gng0 👋🏾, thanks for contributing to this change 🙌🏾 . Can you refactor this part to contain the same href processing we do in the `else` condition? 

```
.....
  if (href) {
    href = locationUtil.assureBaseUrl(href.replace(/\n/g, ''));
    href = replaceVariables(href, dataLinkScopedVars, VariableFormatID.UriEncode);
    href = locationUtil.processUrl(href);
```

**Review by axelavargas — CHANGES_REQUESTED** [review]
Hey @gng0,  I checked the code and noticed some missing href processing in the approach :), I left a comment. 

I will be keeping an eye on this PR, as soon as the change is implemented, I will approve :)

File: packages/grafana-data/src/field/fieldOverrides.ts (hunk 1 above)
Comment (gng0) [reply packages/grafana-data/src/field/fieldOverrides.ts]: Thanks for catching that @axelavargas 
I've just made the change!

**Review by axelavargas — APPROVED** [review]
🥳

**Comment by axelavargas** [comment]
🤔 the CI is failing again, prettier issues:

Output of the CI
> yarn run prettier:check
> [warn] packages/grafana-data/src/field/fieldOverrides.ts
> [warn] Code style issues found in the above file. Run Prettier to fix.

**Comment by gng0** [comment]
@axelavargas @ivanortegaalba just bumping this 😄

**Pull request merged**